pkg/optimizelyjson/optimizely_json.go

```diff
@@ -0,0 +1,98 @@
+/****************************************************************************
+ * Copyright 2020, Optimizely, Inc. and contributors                        *
+ *                                                                          *
+ * Licensed under the Apache License, Version 2.0 (the "License");          *
+ * you may not use this file except in compliance with the License.         *
+ * You may obtain a copy of the License at                                  *
+ *                                                                          *
+ *    http://www.apache.org/licenses/LICENSE-2.0                            *
+ *                                                                          *
+ * Unless required by applicable law or agreed to in writing, software      *
+ * distributed under the License is distributed on an "AS IS" BASIS,        *
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. *
+ * See the License for the specific language governing permissions and      *
+ * limitations under the License.                                           *
+ ***************************************************************************/
+
+// Package optimizelyjson //
+package optimizelyjson
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strings"
+)
+
+// OptimizelyJSON holds the underlying structure of the object
+type OptimizelyJSON struct {
+	jsonRepr map[string]interface{}
+}
+
+// NewOptimizelyJSON constructs the object
+func NewOptimizelyJSON(jsonRepr map[string]interface{}) *OptimizelyJSON {
+	return &OptimizelyJSON{jsonRepr: jsonRepr}
+}
+
+// ToString returns the string representation of json
+func (optlyJson OptimizelyJSON) ToString() (string, error) {
+	jsonBytes, err := json.Marshal(optlyJson.jsonRepr)
+	if err != nil {
+		return "", err
+	}
+
+	return string(jsonBytes), nil
+}
+
+// ToDict returns the native representation of json (map of interface)
+func (optlyJson OptimizelyJSON) ToDict() map[string]interface{} {
```

Comment:
Doesn't it sound confusing to call it `ToDict` while go uses `map` for `key/value` pairs.

Reply:
> Doesn't it sound confusing to call it `ToDict` while go uses `map` for `key/value` pairs.

yeah, I wanted to come up with something generic to all strictly typed languages. But maybe choosing something native to the language is better here

Reply:
Agree `ToMap` makes more sense in this context.